### PR TITLE
Return Single trait for toArray() method

### PR DESCRIPTION
**Pull request by lucas34**

### CODE REVIEW

File: RxSwift/Observables/ToArray.swift [hunk 1]
```diff
@@ -10,17 +10,17 @@
...
     - seealso: [toArray operator on reactivex.io](http://reactivex.io/documentation/operators/to.html)
     
-    - returns: An observable sequence containing all the emitted elements as array.
+    - returns: A Single sequence containing all the emitted elements as array.
     */
     public func toArray()
-        -> Observable<[E]> {
-        return ToArray(source: self.asObservable())
+        -> Single<[E]> {
+        return ToArray(source: self.asObservable()).asSingle()
```
Comment (freak4pc) [inline RxSwift/Observables/ToArray.swift]: Basically I think you can change this one to:

```suggestion
        return PrimitiveSequence(raw: self.asObservable())
```

And it'd be good to go

File: RxSwift/Observables/ToArray.swift [hunk 2]
```diff
@@ -10,17 +10,17 @@
...
     - seealso: [toArray operator on reactivex.io](http://reactivex.io/documentation/operators/to.html)
     
-    - returns: An observable sequence containing all the emitted elements as array.
+    - returns: A Single sequence containing all the emitted elements as array.
     */
     public func toArray()
-        -> Observable<[E]> {
-        return ToArray(source: self.asObservable())
+        -> Single<[E]> {
+        return PrimitiveSequence(raw: self.asObservable())
```
Comment (freak4pc) [inline RxSwift/Observables/ToArray.swift]: That's not what I wrote :) This should be 

```suggestion
        return PrimitiveSequence(raw: ToArray(source: self.asObservable()))
```

Comment (freak4pc) [reply RxSwift/Observables/ToArray.swift]: Oh I did write that, my mistake :O

**Comment by lucas34** [comment]
@freak4pc Thanks for the suggestion ! All checks passed.

**Pull request merged**